cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -168,6 +168,31 @@ char* upper_utf8(int64 context, const char* data, int32 data_len, int32_t* out_l
   return ret;
 }
 
+// Convert a utf8 sequence to lower case.
+// TODO : This handles only ascii characters.
+FORCE_INLINE
+const char* lower_utf8(int64 context, const char* data, int32 data_len,
+                       int32_t* out_len) {
+  char* ret = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context, data_len));
+  if (ret == nullptr) {
```

Comment:
While we are at it, can you add the same error handling to `upper_utf8` above?

Reply:
done. Thanks for reviewing